src/NLog/Targets/DatabaseConnectionInfo.cs

```diff
@@ -0,0 +1,113 @@
+﻿// 
+// Copyright (c) 2004-2020 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__
+
+namespace NLog.Targets
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Data;
+    using System.Globalization;
+    using NLog.Config;
+    using NLog.Internal;
+    using NLog.Layouts;
+
+    /// <summary>
+    /// Information about database command + parameters.
+    /// </summary>
+    [NLogConfigurationItem]
+    public class DatabaseConnectionInfo
```

Comment:
Maybe name it DatabaseConnectionProperty?

Reply:
I want it to match DatabaseParameterInfo

Reply:
- multiple DatabaseParameterInfo means multiple parameters
- multiple DatabaseConnectionInfo doesn't mean multiple connections ;)

So the same name is a bit confusing

Reply:
Not sure I understand. DatabaseConnectionInfo resides in a list. So does DatabaseParameterInfo

Reply:
- **Two** DatabaseParameterInfos will create/work on **two** database parameters. 
- **Two** DatabaseConnectionInfos work on **one single** DBconnection.

The same name won't work well if the don't behave the same. 

Dunno how I could explain this differently.

Reply:
Let me rephrase your sentence:

- **Two** DatabaseParameterInfos will assign two parameters to a DbCommand
- **Two** DatabaseConnectionInfos will assign two properties to a DbConnection

And both names makes sense to me.

Reply:
Maybe it's DatabaseConnectionPropertyInfo then

Reply:
> Let me rephrase your sentence:
> 
> * **Two** DatabaseParameterInfos will assign two parameters to a DbCommand 
> * **Two** DatabaseConnectionInfos will assign two properties to a DbConnection
> 

Don't you see the inconsistency? ParameterInfos/DbCommand  vs ConnectionInfos/DbConnection?

Any way, consistency aside, I find the name  ConnectionInfo confusing

Reply:
Ok lets rename it to DatabasePropertyInfo. Then it can also be reused for DbCommand-properties.

Reply:
Also the same in the target code:

c#
```
  public IList<DatabaseParameterInfo> Parameters { get; } = new List<DatabaseParameterInfo>();
 public IList<DatabaseConnectionInfo> ConnectionProperties { get; } = new List<DatabaseConnectionInfo>();
```

- **Parameter**s  - Database**Parameter**Info
- **ConnectionProperties**  - DatabaseConnectionInfo

Reply:
> Ok lets rename it to DatabasePropertyInfo.

👍 that's a good idea

Reply:
Ooops it became DatabaseObjectPropertyInfo